### PR TITLE
[fix] use new i2c interface for esp32_p4_function_ev_board. (BSP-545)

### PR DESCRIPTION
**Pull request by 100312dog**
# ESP-BSP Pull Request checklist

> Note: For new BSPs create a PR with this [link](https://github.com/espressif/esp-bsp/compare/main...my-branch?quick_pull=1&template=pr_template_bsp.md).
- [x] Version of modified component bumped
- [x] CI passing

# Change description
BSP uses old i2c interface. when another component such as esp-video uses new i2c interface. They will conflict.

### CODE REVIEW

**Comment by tore-espressif** [comment]
@100312dog Thank you for the PR! Could you please bump version in https://github.com/espressif/esp-bsp/blob/master/bsp/esp32_p4_function_ev_board/idf_component.yml to 4.0.0?

cc @espzav

**Review by tore-espressif — APPROVED** [review]

**Comment by espzav** [comment]
HI @100312dog, thank you for this PR. We are discussing solution for support old and new driver for I2C in BSP.

File: bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c [hunk 1]
```diff
@@ -393,8 +387,9 @@ esp_err_t bsp_touch_new(const bsp_touch_config_t *config, esp_lcd_touch_handle_t
         },
     };
     esp_lcd_panel_io_handle_t tp_io_handle = NULL;
-    const esp_lcd_panel_io_i2c_config_t tp_io_config = ESP_LCD_TOUCH_IO_I2C_GT911_CONFIG();
-    ESP_RETURN_ON_ERROR(esp_lcd_new_panel_io_i2c((esp_lcd_i2c_bus_handle_t)BSP_I2C_NUM, &tp_io_config, &tp_io_handle), TAG, "");
+    esp_lcd_panel_io_i2c_config_t tp_io_config = ESP_LCD_TOUCH_IO_I2C_GT911_CONFIG();
+    tp_io_config.scl_speed_hz = CONFIG_BSP_I2C_CLK_SPEED_HZ;
+    ESP_RETURN_ON_ERROR(esp_lcd_new_panel_io_i2c(bsp_i2c_handle, &tp_io_config, &tp_io_handle), TAG, "");
```
Comment (espzav) [inline bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c:392]: Here should be used function `esp_lcd_new_panel_io_i2c_v2`

File: bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c [hunk 2]
```diff
@@ -38,36 +38,30 @@ static lv_indev_t *disp_indev = NULL;
 #endif // (BSP_CONFIG_NO_GRAPHIC_LIB == 0)
 
 sdmmc_card_t *bsp_sdcard = NULL;    // Global uSD card handler
-static bool i2c_initialized = false;
 static TaskHandle_t usb_host_task;  // USB Host Library task
+i2c_master_bus_handle_t bsp_i2c_handle = NULL;   // Global i2c master bus handler
```
Comment (espzav) [inline bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c:42]: Please, can we add get function like this: https://github.com/espressif/esp-bsp/blob/3284337601d3e65d0a60d4a1c4428b4e581e4c67/bsp/esp-box/esp-box.c#L116

and this: https://github.com/espressif/esp-bsp/blob/3284337601d3e65d0a60d4a1c4428b4e581e4c67/bsp/esp-box/include/bsp/esp-box.h#L241

**Review by espzav — COMMENTED** [review]
@100312dog Thank you for this review. I left some comments.

File: bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c (hunk 1 above)
Comment (tore-espressif) [reply bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c:392]: @espzav Actually, the correct v1 or v2 call is selected at build time here https://github.com/espressif/esp-idf/blob/master/components/esp_lcd/include/esp_lcd_io_i2c.h#L118

The `_Generic` keyword select v1 or v2 based on passed type https://www.geeksforgeeks.org/_generic-keyword-c/

Comment (espzav) [reply bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c:392]: Ok, thank you for correct me. Then it is ok.

File: bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c (hunk 2 above)
Comment (100312dog) [reply bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c:42]: It's ok.

Comment (espzav) [reply bsp/esp32_p4_function_ev_board/esp32_p4_function_ev_board.c:42]: @100312dog please, do you plan add mentioned functions?

**Review by espzav — APPROVED** [review]

**Pull request merged**